linter check but otherwise LGTM

> error TS2367: This comparison appears to be unintentional because the types '"text" | "input" | "div" | "image" | "placeholder" | "rectangle" | "web_view" | "radio_group" | "status_bar" | "navigation_bar"' and '"screenshot"' have no overlap.

--- a/ee/frontend/mobile-replay/mobile.types.ts
+++ b/ee/frontend/mobile-replay/mobile.types.ts
@@ -67,6 +67,7 @@ export type serializedNodeWithId = serializedNode & { id: number }
 export type MobileNodeType =
     | 'text'
     | 'image'
+    | 'screenshot'
     | 'rectangle'
     | 'placeholder'
     | 'web_view'
@@ -75,7 +76,6 @@ export type MobileNodeType =
     | 'radio_group'
     | 'status_bar'
     | 'navigation_bar'
-    | 'screenshot'
 
 export type MobileStyles = {
     /**
@@ -304,6 +304,7 @@ export type wireframeNavigationBar = wireframeBase & {
 export type wireframe =
     | wireframeText
     | wireframeImage
+    | wireframeScreenshot
     | wireframeRectangle
     | wireframeDiv
     | wireframeInputComponent
@@ -312,7 +313,6 @@ export type wireframe =
     | wireframePlaceholder
     | wireframeStatusBar
     | wireframeNavigationBar
-    | wireframeScreenshot
 
 // the rrweb full snapshot event type, but it contains wireframes not html
 export type fullSnapshotEvent = {

--- a/ee/frontend/mobile-replay/schema/mobile/rr-mobile-schema.json
+++ b/ee/frontend/mobile-replay/schema/mobile/rr-mobile-schema.json
@@ -278,6 +278,7 @@
             "enum": [
                 "text",
                 "image",
+                "screenshot",
                 "rectangle",
                 "placeholder",
                 "web_view",
@@ -285,8 +286,7 @@
                 "div",
                 "radio_group",
                 "status_bar",
-                "navigation_bar",
-                "screenshot"
+                "navigation_bar"
             ],
             "type": "string"
         },
@@ -384,6 +384,9 @@
                     "$ref": "#/definitions/wireframeImage"
                 },
                 {
+                    "$ref": "#/definitions/wireframeScreenshot"
+                },
+                {
                     "$ref": "#/definitions/wireframeRectangle"
                 },
                 {
@@ -406,9 +409,6 @@
                 },
                 {
                     "$ref": "#/definitions/wireframeNavigationBar"
-                },
-                {
-                    "$ref": "#/definitions/wireframeScreenshot"
                 }
             ]
         },
@@ -992,7 +992,7 @@
         },
         "wireframeScreenshot": {
             "additionalProperties": false,
-            "description": "a screenshot behaves exactly like an image, but it is expected to be a screenshot of the screen at the time of the event. When sent as a mutation it must always attached to the root of the playback, when sent as an initial snapshot it must be sent as the first or only node so that it attaches to the body of the playback",
+            "description": "a screenshot behaves exactly like an image, but it is expected to be a screenshot of the screen at the time of the event, when sent as a mutation it must always attached to the root of the playback, when sent as an initial snapshot it must be sent as the first or only snapshot so that it attaches to the body of the playback",
             "properties": {
                 "base64": {
                     "description": "this will be used as base64 encoded image source, with no other attributes it is assumed to be a PNG, if omitted a placeholder is rendered",

--- a/ee/frontend/mobile-replay/transformer/transformers.ts
+++ b/ee/frontend/mobile-replay/transformer/transformers.ts
@@ -992,6 +992,10 @@ function isMobileIncrementalSnapshotEvent(x: unknown): x is MobileIncrementalSna
     return hasMutationSource && (hasAddedWireframe || hasUpdatedWireframe)
 }
 
+function chooseParentId(nodeType: MobileNodeType, providedParentId: number): number {
+    return nodeType === 'screenshot' ? BODY_ID : providedParentId
+}
+
 function makeIncrementalAdd(add: MobileNodeMutation, context: ConversionContext): addedNodeMutation[] | null {
     const converted = convertWireframe(add.wireframe, context)
 
@@ -1000,7 +1004,7 @@ function makeIncrementalAdd(add: MobileNodeMutation, context: ConversionContext)
     }
 
     const addition: addedNodeMutation = {
-        parentId: add.wireframe.type === 'screenshot' ? BODY_ID : add.parentId,
+        parentId: chooseParentId(add.wireframe.type, add.parentId),
         nextId: null,
         node: converted.result,
     }
@@ -1018,7 +1022,7 @@ function makeIncrementalAdd(add: MobileNodeMutation, context: ConversionContext)
  */
 function makeIncrementalRemoveForUpdate(update: MobileNodeMutation): removedNodeMutation {
     return {
-        parentId: update.wireframe.type === 'screenshot' ? BODY_ID : update.parentId,
+        parentId: chooseParentId(update.wireframe.type, update.parentId),
         id: update.wireframe.id,
     }
 }

